src/character.cpp

```diff
@@ -812,8 +812,8 @@ int Character::CanHoldItem(short itemid, int max_amount)
 {
 	int amount = max_amount;
 
-	if (int(this->world->config["EnforceWeight"]) >= 2
-	 && SourceDutyAccess() < static_cast<int>(world->config["unlimitedweight"]))
+	if (int(this->world->config["EnforceWeight"]) == 0 ||
```

Comment:
I'm not entirely sure what the representation of `EnforceWeight` is intended to be. The comment in the config says 0 == no enforcement, 1 == attack block, 2 == pickup block. I'm assuming if you're attack blocked you're also blocked from picking up items, but not entirely sure if that's how it should be.

Reply:
As far as I can tell, the logic of this check was fine and shouldn't be inverted like this.
The problem is that `unlimitedweight` is in the `admin.ini` config and not the regular config. `static_cast<int>(world->config["unlimitedweight"])` will evaluate to 0, causing `SourceDutyAccess() < static_cast<int>(world->config["unlimitedweight"])` to evaluate to false.

Reply:
I didn't even consider that the setting was in admin config. I'll put it back the way it was and have it check the admin config properly